### PR TITLE
Update WIP / Draft check to check for skip prefix in commit messages

### PR DESCRIPTION
**Pull request by jameesjohn**
## Explanation
Update WIP / Draft check to check for skip prefix in commit messages

## Checklist
- [x] I have successfully deployed my own instance of Oppiabot.
  - You can find instructions for doing this [here](https://github.com/oppia/oppiabot/wiki/Deploying-your-own-instance-of-the-oppiabot).
- [x] I have manually tested all the changes made in this PR following the [manual tests matrix](https://github.com/oppia/oppiabot/wiki/Manual-Tests-Matrix).

### CODE REVIEW

File: spec/checkWipDraftPRSpec.js [hunk 1]
```diff
@@ -66,8 +66,15 @@ describe('Oppiabot\'s', () => {
     done();
   });
 
-  describe('WIP PRs', () => {
+  describe('WIP PRs without skip prefix', () => {
     beforeEach((done) => {
+      github.git = {
+        getCommit: jasmine.createSpy('getCommit').and.resolveTo({
+          data: {
+            message: 'changes'
```
Comment (ankita240796) [inline spec/checkWipDraftPRSpec.js:74]: Make it more relevant like `commit withou skip prefix`

File: spec/checkWipDraftPRSpec.js [hunk 2]
```diff
@@ -117,16 +133,63 @@ describe('Oppiabot\'s', () => {
     });
   });
 
+  describe('WIP PRs with skip prefix', () => {
+    beforeEach((done) => {
+      github.git = {
+        getCommit: jasmine.createSpy('getCommit').and.resolveTo({
+          data: {
+            message: '[ci skip] changes'
```
Comment (ankita240796) [inline spec/checkWipDraftPRSpec.js:141]: Ditto.

File: spec/checkWipDraftPRSpec.js [hunk 3]
```diff
@@ -135,12 +198,19 @@ describe('Oppiabot\'s', () => {
 
     it('should check when PR is reopnend', async() => {
       spyOn(checkWipDraftPRModule, 'checkWIP').and.callThrough();
-
+      github.git = {
+        getCommit: jasmine.createSpy('getCommit').and.resolveTo({
+          data: {
+            message: 'changes'
```
Comment (ankita240796) [inline spec/checkWipDraftPRSpec.js:204]: Ditto.

File: spec/checkWipDraftPRSpec.js [hunk 4]
```diff
@@ -150,6 +220,13 @@ describe('Oppiabot\'s', () => {
 
   describe('Draft PRs', () => {
     beforeEach((done) => {
+      github.git = {
+        getCommit: jasmine.createSpy('getCommit').and.resolveTo({
+          data: {
+            message: 'changes'
```
Comment (ankita240796) [inline spec/checkWipDraftPRSpec.js:226]: Ditto.

File: spec/checkWipDraftPRSpec.js [hunk 5]
```diff
@@ -207,8 +293,65 @@ describe('Oppiabot\'s', () => {
     });
   });
 
+  describe('Draft PRs with skip prefix', () => {
+    beforeEach((done) => {
+      github.git = {
+        getCommit: jasmine.createSpy('getCommit').and.resolveTo({
+          data: {
+            message: '[skip ci] changes'
```
Comment (ankita240796) [inline spec/checkWipDraftPRSpec.js:301]: Ditto.

File: spec/checkWipDraftPRSpec.js [hunk 6]
```diff
@@ -207,8 +293,65 @@ describe('Oppiabot\'s', () => {
...
+      expect(github.issues.update).not.toHaveBeenCalled();
+    });
+  });
+
   describe('Neither Draft nor WIP PRs', () => {
     beforeEach((done) => {
+      github.git = {
+        getCommit: jasmine.createSpy('getCommit').and.resolveTo({
+          data: {
+            message: '[skip ci] changes'
```
Comment (ankita240796) [inline spec/checkWipDraftPRSpec.js:351]: Ditto.

File: lib/checkWipDraftPR.js [hunk 7]
```diff
@@ -41,26 +61,28 @@ module.exports.checkWIP = async (context) => {
...
-    const commentParams = context.repo({
-      issue_number: pullRequest.number,
-      body: commentBody,
-    });
-    await context.github.issues.createComment(commentParams);
+    const hasSkipCIMessage = await isSkipCICommit(context);
+    if (!hasSkipCIMessage) {
+      const link = 'here'.link(
+        'https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia' +
+        '#wip--draft-pull-requests');
```
Comment (ankita240796) [inline lib/checkWipDraftPR.js:68]: Have you updated this documentation as well?

File: spec/checkWipDraftPRSpec.js [hunk 8]
```diff
@@ -91,10 +108,9 @@ describe('Oppiabot\'s', () => {
         '#wip--draft-pull-requests');
       const author = pullRequestEditedPayload.payload.pull_request.user.login;
       const commentBody = (
-        'Hi @' + author + ', WIP/Draft PRs are highly discouraged. You can ' +
-        'learn more about it ' + link + '. You can reopen it when it\'s ' +
-        'ready to be reviewed and ensure that it is without any WIP phrase ' +
-        'in title or body. Thanks!');
+        'Hi @' + author + ', when creating WIP/Draft PRs, ensure that ' +
+        'your commit messages are prefixed with **[ci skip]** to prevent CI ' +
```
Comment (ankita240796) [inline spec/checkWipDraftPRSpec.js:112]: [skip ci] is allowed as well.

**Comment by ankita240796** [comment]
The frontend tests are failing, can you check if your branch is updated with master? The failure looks like the older one which I have already fixed.

File: lib/checkWipDraftPR.js (hunk 7 above)
Comment (jameesjohn) [reply lib/checkWipDraftPR.js:68]: Yes I did https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#wip--draft-pull-requests

File: spec/checkWipDraftPRSpec.js (hunk 1 above)
Comment (jameesjohn) [reply spec/checkWipDraftPRSpec.js:74]: Done.

File: spec/checkWipDraftPRSpec.js (hunk 8 above)
Comment (jameesjohn) [reply spec/checkWipDraftPRSpec.js:112]: Updated.

File: spec/checkWipDraftPRSpec.js (hunk 2 above)
Comment (jameesjohn) [reply spec/checkWipDraftPRSpec.js:141]: Done.

File: spec/checkWipDraftPRSpec.js (hunk 3 above)
Comment (jameesjohn) [reply spec/checkWipDraftPRSpec.js:204]: Done.

File: spec/checkWipDraftPRSpec.js (hunk 4 above)
Comment (jameesjohn) [reply spec/checkWipDraftPRSpec.js:226]: Done.

File: spec/checkWipDraftPRSpec.js (hunk 5 above)
Comment (jameesjohn) [reply spec/checkWipDraftPRSpec.js:301]: Done.

File: spec/checkWipDraftPRSpec.js (hunk 6 above)
Comment (jameesjohn) [reply spec/checkWipDraftPRSpec.js:351]: Done.

**Comment by jameesjohn** [comment]
PTAL @ankita240796, I've addressed all comments.
Thanks!

File: lib/checkWipDraftPR.js (hunk 7 above)
Comment (ankita240796) [reply lib/checkWipDraftPR.js:68]: Add a link to the travis and circle ci documentation as well in the wiki:

1. https://docs.travis-ci.com/user/customizing-the-build/#skipping-a-build
2.  https://circleci.com/docs/2.0/skip-build/#skipping-a-build

**Comment by jameesjohn** [comment]
Done. https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#wip--draft-pull-requests

**Comment by ankita240796** [comment]
Great, Thanks @jameesjohn!

**Pull request merged**